crates/tap-agent/src/agent/sender_allocation.rs

```diff
@@ -93,30 +89,32 @@ pub enum RavError {
     #[error("All receipts are invalid")]
     AllReceiptsInvalid,
 
-    #[error("Receipt is not legacy")]
-    ReceiptNotCompatible,
-
     #[error(transparent)]
     Other(#[from] anyhow::Error),
 }
 
-type TapManager = tap_core::manager::Manager<TapAgentContext, TapReceipt>;
+type TapManager<T> = tap_core::manager::Manager<TapAgentContext<T>, TapReceipt>;
 
 /// Manages unaggregated fees and the TAP lifecyle for a specific (allocation, sender) pair.
-pub struct SenderAllocation;
+pub struct SenderAllocation<T>(PhantomData<T>);
+impl<T: NetworkVersion> Default for SenderAllocation<T> {
+    fn default() -> Self {
+        Self(PhantomData)
+    }
+}
```

Comment:
Awesome you're using these patterns but they're not that readable and look kind of ridiculous, so let's add some documentation to explain what the PURPOSE is 🙏

Reply:
added docs